src/components/Sep8Send/Sep8Approval.tsx

```diff
@@ -59,39 +55,24 @@ export const Sep8Send = () => {
         approvalServer,
       };
 
-      dispatch(sep8SendPaymentAction(params));
+      dispatch(sep8ReviseTransactionAction(params));
     }
   };
 
-  const handleCloseModal = useCallback(() => {
-    setSep8PaymentModalVisible(false);
+  const handleCloseModal = () => {
     resetFormState();
-    dispatch(resetActiveAssetAction());
-    dispatch(resetSep8SendAction());
-  }, [dispatch]);
+    onClose();
+  };
 
   // use effect
   useEffect(() => {
-    if (sep8Send.status === ActionStatus.CAN_PROCEED) {
-      setSep8PaymentModalVisible(true);
-    }
-
-    if (sep8Send.status === ActionStatus.SUCCESS && account.data?.id) {
-      dispatch(
-        fetchAccountAction({
-          publicKey: account.data.id,
-          secretKey: account.secretKey,
-        }),
-      );
-      handleCloseModal();
+    if (
+      sep8Send.status === ActionStatus.CAN_PROCEED &&
+      sep8Send.data.revisedTransaction.revisedTxXdr
```

Comment:
I'm assuming we'll always have `sep8Send.data.revisedTransaction` (no need to do optional chaining)?

Reply:
That's right, yes.